mark.c

```diff
@@ -810,23 +816,38 @@ GC_INNER mse * GC_mark_from(mse *mark_stack_top, mse *mark_stack,
         for(;;) {
           PREFETCH(limit - PREF_DIST*CACHE_LINE_SIZE);
           GC_ASSERT((word)limit >= (word)current_p);
-          deferred = *(word *)limit;
-          FIXUP_POINTER(deferred);
-          limit -= ALIGNMENT;
-          if (deferred >= (word)least_ha && deferred < (word)greatest_ha) {
-            PREFETCH((ptr_t)deferred);
-            break;
-          }
+          if( cheri_getaddress(limit) < cheri_getbase(limit)
+              || cheri_getaddress(limit) >= (cheri_getbase(limit) + cheri_getlength(limit))
+              || cheri_gettag(limit) == 0) {
+	    limit -= ALIGNMENT; 
+	  } else {
+            deferred = *(word *)limit;
+            FIXUP_POINTER(deferred);
+	    limit -= ALIGNMENT; 
```

Comment:
Indentation issue.

Reply:
fixed in commit [408f667b](https://github.com/capablevms/bdwgc/pull/1/commits/408f667bb59d4d389e44ff70d5631bf922413ed9)